extensions/src/common/extension/decorators/newBlocks.ts

```diff
@@ -121,20 +52,28 @@ interface TemplateEngine<TBlockType> {
         const Args extends any[],
         const Return,
     >
-    (
-        strings: TemplateStringsArray, ...args: Argument.MapToScratch<Args>
-    ): TypedMethodDecorator<This, Args, Return, Utility.Method<This, Args, Return>>;
+        (
+            strings: TemplateStringsArray, ...args: Argument.MapToScratch<Args>
+        ): TypedMethodDecorator<This, Args, Return, Utility.Method<This, Args, Return>>;
 
     /**
      * 
      */
     execute<
-            const This extends ExtensionInstance,
-            const Args extends any[],
-            const Return,
-        >
-    (
-        builder: (instance: This, tag: Utility.TaggedTemplate<Argument.MapToScratch<Args>, TypedMethodDecorator<This, Args, Return, Utility.Method<This, Args, Return>>>) 
-        => TypedMethodDecorator<This, Args, Return, Utility.Method<This, Args, Return>>
-    );
+        const This extends ExtensionInstance,
+        Args extends any[],
+        Return,
+    >
+        (
+            builder: (
+                instance: This,
+                tag: Utility.TaggedTemplate<Argument.MapToScratch<Args>, BlockMetadata<(...args: Args) => Return>>
+            ) => BlockMetadata<(...args: Args) => Return>
+        ): TypedMethodDecorator<This, Args, Return, Utility.Method<This, Args, Return>>;
```

Comment:
I think the nesting of angle boys and what not got things confused here, as well as the multiple levels of functions-as-arguments.

This `execute` function type definition takes a single argument, which is a function called "builder". "builder" is a function that takes two arguments, the second one being "tag". "tag" is a function that takes two arguments ... etc. etc. Importantly, the `execute` function is the thing that returns the `TypedMethodDecorator`